go/libraries/doltcore/sqle/dfunctions/dolt_commit.go

```diff
@@ -0,0 +1,224 @@
+		t, err := time.Parse(layout, dateStr)
+
+		if err == nil {
+			return t, nil
+		}
+	}
+
+	return time.Time{}, errors.New("error: '" + dateStr + "' is not in a supported format.")
+}
+
+func parseAuthor(authorStr string) (string, string, error) {
+	if len(authorStr) == 0 {
+		return "", "", errors.New("Option 'author' requires a value")
+	}
+
+	reg := regexp.MustCompile("(?m)([^)]+) \\<([^)]+)") // Regex matches Name <email
+	matches := reg.FindStringSubmatch(authorStr)        // This function places the original string at the beginning of matches
+
+	// If name and email are provided
+	if len(matches) != 3 {
+		return "", "", errors.New("Author not formatted correctly. Use 'Name <author@example.com>' format")
+	}
+
+	name := matches[1]
+	email := strings.ReplaceAll(matches[2], ">", "")
+
+	return name, email, nil
+}
+
+// Trims the double quotes for the param.
+func trimQuotes(param string) string {
+	if len(param) > 0 && param[0] == '"' {
+		param = param[1:]
+	}
+
+	if len(param) > 0 && param[len(param)-1] == '"' {
+		param = param[:len(param)-1]
+	}
+
+	return param
+}
+
+func (d DoltCommitFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	// Get the information for the sql context.
+	dbName := ctx.GetCurrentDatabase()
+	dSess := sqle.DSessFromSess(ctx.Session)
+
+	doltdb, ok := dSess.GetDoltDB(dbName)
+
+	if !ok {
+		return nil, fmt.Errorf("Could not load %s", dbName)
+	}
+
+	rsr, ok := dSess.GetDoltDBRepoStateReader(dbName)
+
+	if !ok {
+		return nil, fmt.Errorf("Could not load the %s RepoStateReader", dbName)
+	}
+
+	rsw, ok := dSess.GetDoltDBRepoStateWriter(dbName)
+
+	if !ok {
+		return nil, fmt.Errorf("Could not load the %s RepoStateWriter", dbName)
+	}
+
+	ap := createArgParser()
+
+	// Get the args for DOLT_COMMIT.
+	args := make([]string, 1)
+	for i := range d.children {
+		temp := d.children[i].String()
+		str := trimQuotes(temp)
+		args = append(args, str)
+	}
+
+	apr := cli.ParseArgs(ap, args, nil)
+
+	// Parse the author flag. Return an error if not.
+	var name, email string
+	var err error
+	if authorStr, ok := apr.GetValue(authorParam); ok {
+		name, email, err = parseAuthor(authorStr)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		name = dSess.Username
+		email = dSess.Email
+	}
+
+	// Get the commit message.
+	msg, msgOk := apr.GetValue(commitMessageArg)
+	if !msgOk {
+		return nil, fmt.Errorf("Must provide commit message.")
+	}
+
+	// Specify the time if the date parameter is not.
+	t := time.Now()
+	if commitTimeStr, ok := apr.GetValue(dateParam); ok {
+		var err error
+		t, err = parseDate(commitTimeStr)
+
+		if err != nil {
+			return nil, fmt.Errorf(err.Error())
+		}
+	}
+
+	err = actions.CommitStaged(ctx, doltdb, rsr, rsw, actions.CommitStagedProps{
+		Message:          msg,
+		Date:             t,
+		AllowEmpty:       apr.Contains(allowEmptyFlag),
+		CheckForeignKeys: !apr.Contains(forceFlag),
+		Name:             name,
+		Email:            email,
+	})
+
+	return "Staged", err
```

Comment:
Expression should evaluate to the hash of the new commit. Should have a test of this.

Reply:
Do we have a bats manner for comparing hashes? Or do you want a go test?